frontend/src/pages/ChainMemberList.tsx

```diff
@@ -54,73 +63,71 @@ const ChainMemberList = () => {
     }
   };
 
+
+  //pagination
+  const handleChangePage = (e: any, newPage: any) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (e: any) => {
+    setRowsPerPage(+e.target.value);
+    setPage(0);
+  };
+
   useEffect(() => {
     (async () => {
       try {
         const chainData = await getChain(chainId);
         setChain(chainData);
-        const users = await getUsersForChain(chainId);
-        setUsers(users);
+        const chainUsers = await getUsersForChain(chainId);
+        setUsers(chainUsers);
         setPublishedValue({ published: chainData.published });
+
+        const adminData: any = users?.filter((user: IUser) => {
+          if (user.role === "chainAdmin") {
+            return setAdmin(user);
```

Comment:
I'm pretty sure this is buggy. You should return a bool here, not whatever setAdmin returns. Also, you probably shouldn't execute code with side-effects inside the callback. Use find instead of filter: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

`setAdmin(users?.filter((user: IUser) => user.role === "chainAdmin"));`